pandas/core/series.py

```diff
@@ -4303,6 +4304,54 @@ def between(self, left, right, inclusive=True) -> "Series":
 
         return lmask & rmask
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _convert_dtypes(
+        self: ABCSeries,
+        infer_objects: bool = True,
+        convert_string: bool = True,
+        convert_integer: bool = True,
+        convert_boolean: bool = True,
+    ) -> ABCSeries:
+        """
+        Convert objects to best possible type, and optionally,
+        columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        Parameters
+        ----------
+        infer_objects : bool, default True
+            Whether object dtypes should be converted to the best possible types.
+        convert_string : bool, default True
+            Whether object dtypes should be converted to ``StringDtype()``.
+        convert_integer : bool, default True
+            Whether, if possible, conversion can be done to integer extension types.
+        convert_boolean : bool, defaults True
+            Whether object dtypes should be converted to ``BooleanDtypes()``.
+
+        Returns
+        -------
+        Series
+            copy of Series with new (or existing) dtype
+        """
+        input_series = self
+        if infer_objects:
+            input_series = input_series.infer_objects()
+            if is_object_dtype(input_series):
+                input_series = input_series.copy()
```

Comment:
why this copy?

Reply:
I am making sure that `convert_dtypes` *always* returns a copy.  If `infer_objects()` is called and keeps the dtype as `object` then you have to make the copy, because `infer_objects` will not return a copy in that case.

Reply:
`.infer_objects` *always* copies (it doesn't actually expose the paramater)

Reply:
No, it doesn't:
```python
>>> s=pd.Series(["a", "b"])
>>> s
0    a
1    b
dtype: object
>>> s2=s.infer_objects()
>>> s2
0    a
1    b
dtype: object
>>> s2[0] = "c"
>>> s2
0    c
1    b
dtype: object
>>> s
0    c
1    b
dtype: object
```
Note that `s2` is the result of `infer_objects`, but when you change `s2`, it changes `s`

Reply:
hmm, ok, that's likely a reason we need a ``copy=True`` there but that's a separate issue